.github/workflows/refresh_dist.yml

```
@@ -12,28 +12,40 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - id: changed-files
-        uses: tj-actions/changed-files@v44
+      - id: changed-package-json
+        uses: tj-actions/changed-features@v44
+        with:
+          files: |
+            **/package.json
+            **/package-lock.json
+
+      - id: changed-features
+        uses: tj-actions/changed-features@v44
         with:
           files: |
             features/**/*.yml
             features/**/*.yml.dist
 
       - uses: actions/setup-node@v4
-        if: steps.changed-files.outputs.any_changed == 'true'
+        if: steps.changed-features.outputs.any_changed == 'true'
 
       - run: npm install
-        if: steps.changed-files.outputs.any_changed == 'true'
+        if: steps.changed-features.outputs.any_changed == 'true'
 
-      - name: Refresh dist files
-        if: steps.changed-files.outputs.any_changed == 'true'
+      - name: Refresh dist files for feature changes
+        if: steps.changed-package-json.outputs.any_changed == 'false' && steps.changed-features.outputs.any_changed == 'true'
         env:
-          CHANGED_FEATURES: ${{ steps.changed-files.outputs.all_changed_files }}
+          CHANGED_FEATURES: ${{ steps.changed-features.outputs.all_changed_files }}
         run: |
           npm run dist -- ${CHANGED_FEATURES}
 
+      - name: Refresh all dist files (for package*.json changes)
+        if: steps.changed-package-json.outputs.any_changed == 'true'
+        run: |
+          npm run dist -- features/**/*.yml.dist
```

Is there any reason we can't just update everything and not care about what files were changed?

```suggestion
          npm run dist
```

---

In the "changed-features" case, we don't want to change dists for files not already in-scope for the PR. _Ordinarily_ it shouldn't make a difference, but I was thinking of some of the bad merges that have taken place, where refreshing the dist would add surprising files to the diff of unrelated PRs.

In the "changed-package-json" case, I forgot we changed the dist script to do this by default. I'll fix this.